### DISCUSSION TITLE
Does Jpeg format need comment marker support?

### DISCUSSION THREAD

**Discussion by br3aker** [SixLabors/ImageSharp#2061] [Ideas] [answered]
Current jpeg implementation lacks of COM marker support, does it need to support it? Should be easy to implement as this marker is just an array of bytes - itu spec leaves 'interpretation to the application', decoding API should not try to decode it as it can be anything from literal byte array to some weird text encoding.

So it can be a 'good first issue' or I can work on it a bit later.

**Comment by brianpopow** [comment, marked as answer]
@br3aker I think it would be nice to have support for decoding COM markers and a "good first issue". 

Don't you think it would make sense to add a default interpretation of the data as ASCII, which would make it a bit more useful? Would there be any harm in doing so? I would be also fine with just providing it as a byte array and the user should take care of it, but I just think a optional default interpretation would be a bit ore convenient.

**Reply by br3aker** [reply]
Not sure how to implement such default behaviour from API point of view. Doing it by byte array can be as simple as `IColletion<byte[]>` backed by a `List<byte[]>`. Making multiple implementations via template method, for example, would oblige us to add `GetComment(index)`, `SetComment(index)` and `ClearComments()` because we can no longer return just `ICollection` with all available methods already implemented to the end user. 

Considering COM marker is a very niche thingy of the jpeg images and that usually COM markers contain ASCII data we can go with `char` - more advaced users with special use cases can just do `jpegMetadata.Comments[i].AsSpan().Cast<char, SomeTargetType>()`.

**Reply by brianpopow** [reply]
> Considering COM marker is a very niche thingy of the jpeg images and that usually COM markers contain ASCII data we can go with char - more advaced users with special use cases can just do jpegMetadata.Comments[i].AsSpan().Cast<char, SomeTargetType>().

That sounds good to me 👍

**Reply by br3aker** [reply]
Should I open an issue or unanswered disscussion is enough?

**Reply by JimBobSquarePants** [reply]
I've created an issue from the discussion

**Comment by prabhavmehra** [comment]
@br3aker Did you get a chance to work on this? I was looking for a similar functionality? if not, can you share some api design/implementation notes and I can probably have a look.

**Reply by br3aker** [reply]
Left a comment in the linked issue :)
